File: hira/resources/cloudwatch.py
import boto3

class CloudWatchMetrices:
    def __init__(self):
        # calling a function of cloudwatch
        self.client = boto3.client('cloudwatch')

    def putmetric(self, namespace,metric_name, dimensions, value):
        metric_call = self.client.put_metric_data(
            Namespace=namespace,
            MetricData=[
                {
                    'MetricName': metric_name,

                    'Dimensions': dimensions,       # URL and url value

                    'Value': value,         # Datapoints of url
                }
            ]
        )
        return metric_call

File: hira4sprint/cdk.out/asset.a84ec07360a4863f6caec1a6af96d185e376e07d141099b2e6d561527c7096a9/cloudwatch.ts
var AWS = require('aws-sdk');
export async function publish_metric(namespace:string,metricname:string,url:string,value:number){

    // Create CloudWatch service object
    var cw = new AWS.CloudWatch();

    /*
        metric_Params 
         
        MetricName -> name of metric
        Dimensions -> Key value pair
        Value -> Datapoints
    */

    var metric_Params = {
        MetricData: [
          {
            MetricName: metricname,
            Dimensions: [{ Name: "URL",    Value: url  },],
            Value: value
          },
        ],
        Namespace: namespace
      };
      
      // Putting Data into putMetricData function 
      cw.putMetricData(metric_Params, function(err:any, data:any) {
        if (err) {
          console.log("Error", err);
        } else {
          console.log("Success", JSON.stringify(data));
        }
      });
}